content/blog/using-mcps-to-run-terraform.md

```
@@ -0,0 +1,205 @@
+<br>
+
+### Debugging terraform apply
+At this point, I was ready for the agent to run terraform plan and apply my changes. And here is where the rubber meets the road in using MCPs 🙂. Here’s what I ran into:
+
+**The first issue** was a permissions error — `admin_user` couldn’t create Postgres roles. This issue was totally my fault — I hadn’t granted it any meaningful privileges. I had the MCP repeatedly run `terraform apply`, but either the MCP didn't log/surface the error or Cursor’s Agent Mode wasn't able to interepret the cause of the error.
+![First issue - MCP able to run terraform apply][/img/updates/using-mcps-to-run-terraform/debug1.png]
+
+Eventually, I switched to running `terraform apply` directly in the terminal and got a helpful error: `admin_user` lacked the required privileges. I jumped into Postgres on the command line via `psql` and granted it `CREATEROLE` and `CREATEDB`.
+
+**The second issue** was with the provider config — even after fixing the role permissions, `terraform apply` kept failing when run via the MCP (see screenshots). Again, the specific terraform error wasn’t surfaced.
+![MCP doesn't show the acutual error][/img/updates/using-mcps-to-run-terraform/debug2.png]
```

markdownlint-cli2 (0.17.2)

Reference links and images should use a label that is defined
Missing link or image reference definition: "/img/updates/using-mcps-to-run-terraform/debug2.png"

(MD052, reference-links-images)